### ISSUE TITLE
Avoiding bottlenecks

### ISSUE THREAD

**Issue by kokujin** [fastify/fastify#1390]
I am not sure if this is the right place to ask. Please accept my apologies if it's the wrong place.

We intend to use Fastify for our applications. These applications request data from REST API applications built using [Feathers](https://github.com/feathersjs/feathers).

What would be the best way to keep our Fastify applications fast even if they request data from slower API servers? Thanks.

**Comment by mcollina** [comment]
Feathers provides a REST API layer. Why would you need to wrap a Fastify application on top?

> What would be the best way to keep our Fastify applications fast even if they request data from slower API servers?

You should add some in-process or shared caching for these APIs. As an example, you can store these results in Redis or use fastify-caching.

**Comment by mcollina** [comment]
Then I do not understand your question. Can you make an example?

**Comment by kokujin** [comment]
An example would involve to much to show here. The idea is simple though. Let me try to explain again.

We have applications based on ExpressJS that rendered data based on what they received from an API server. The API server is based on Feathers.

We thought we can achieve better performance values by using Fastify instead of ExpressJS for the client applications I mentioned above.

**Comment by delvedor** [comment]
@kokujin if I understood correctly you are using a [backend for frontend](https://samnewman.io/patterns/architectural/bff/) architecture?

**Comment by kokujin** [comment]
not quite @delvedor, based on the link you posted( thanks by the way, I will place an order for the book). We have an API gateway pattern and not a BFF pattern wich tends to different types of clients.
